cli_li3ds/import_json.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import json
+import pathlib
+
+from cliff.command import Command
+
+from . import api
+
+
+class ImportJson(Command):
+    """ import one or several images
+    """
+
+    log = logging.getLogger(__name__)
+
+    image_date_cache = {}
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def get_parser(self, prog_name):
+        self.log.debug(prog_name)
+        parser = super().get_parser(prog_name)
+        api.add_arguments(parser)
+        parser.add_argument(
+            '--json-dir', '-f', default='.',
+            help='base directory to search for json files (optional, default is ".")')
+        parser.add_argument(
+            '--uri', default='{uri}',
+            help='uri pattern for datasources (optional, default is path through "{uri}")')
+        parser.add_argument(
+            'filename', nargs='+',
+            help='JSON file names, may be Unix-style patterns (e.g. *.json)')
+        return parser
+
+    def take_action(self, parsed_args):
+        server = api.ApiServer(parsed_args, self.log)
+        json_dir = pathlib.Path(parsed_args.json_dir)
+        uri = parsed_args.uri
+
+        objs = api.ApiObjs(server)
+
+        for filename in parsed_args.filename:
+            for json_path in json_dir.rglob(filename):
+                self.log.info('Importing {}'.format(json_path.relative_to(json_dir)))
+                sensors = self.handle_json(objs, json_path, uri, api.Sensor)
+                referentials = self.handle_json(objs, json_path, uri, api.Referential, sensors)
+                ttypes = self.handle_json(objs, json_path, uri, api.TransfoType, referentials)
+                transfos = self.handle_json(objs, json_path, uri, api.Transfo, referentials, ttypes)
+                transfotrees = self.handle_json(objs, json_path, uri, api.Transfotree, transfos)
+                platforms = self.handle_json(objs, json_path, uri, api.Platform)
+                self.handle_json(objs, json_path, uri, api.Config, platforms, transfotrees)
+                projects = self.handle_json(objs, json_path, uri, api.Project)
+                sessions = self.handle_json(objs, json_path, uri, api.Session, projects, platforms)
+                self.handle_json(objs, json_path, uri, api.Datasource, sessions, referentials)
+
+        objs.get_or_create()
+        self.log.info('Success!\n')
+
+    @classmethod
+    def handle_json(cls, objs, json_path, uri, class_, deps1=None, deps2=None):
+        with json_path.open(encoding="iso-8859-1") as f:
```

Comment:
iso-8859-1? The json files are guaranteed to be iso-8859-1 encoded?

Reply:
I just know importing https://github.com/LI3DS/cli-li3ds/blob/import-json/data/2017_17FA7506_C_6.json did not work without specifying this encoding. 
Do you have a better proposition ?